frontend/src/index.test.js

```diff
@@ -1,37 +1,2524 @@
+            "domain_cnt": 1,
+            "domain": "condition_occurrence",
+            "total_cnt": 800,
+            "distinct_person_cnt": "200",
+            "isItem": true,
+            "status": "Def"
+        },
+        {
+            "concept_id": 46269850,
+            "concept_name": "Fistula of large intestine due to chronic ulcerative rectosigmoiditis",
+            "domain_id": "Condition",
+            "vocabulary_id": "SNOMED",
+            "concept_class_id": "Clinical Finding",
+            "standard_concept": "S",
+            "concept_code": "1085251000119106",
+            "invalid_reason": null,
+            "domain_cnt": 1,
+            "domain": "condition_occurrence",
+            "total_cnt": 552,
+            "distinct_person_cnt": "106",
+            "isItem": true,
+            "status": "Def"
+        },
+        {
+            "concept_id": 46269851,
+            "concept_name": "Rectal hemorrhage due to chronic ulcerative rectosigmoiditis",
+            "domain_id": "Condition",
+            "vocabulary_id": "SNOMED",
+            "concept_class_id": "Clinical Finding",
+            "standard_concept": "S",
+            "concept_code": "1085271000119102",
+            "invalid_reason": null,
+            "domain_cnt": 1,
+            "domain": "condition_occurrence",
+            "total_cnt": 60946,
+            "distinct_person_cnt": "5194",
+            "isItem": true,
+            "status": "Def"
+        },
+        {
+            "concept_id": 46269951,
+            "concept_name": "Abscess of intestine co-occurrent and due to ulcerative colitis",
+            "domain_id": "Condition",
+            "vocabulary_id": "SNOMED",
+            "concept_class_id": "Clinical Finding",
+            "standard_concept": "S",
+            "concept_code": "1092841000119100",
+            "invalid_reason": null,
+            "domain_cnt": 1,
+            "domain": "condition_occurrence",
+            "total_cnt": 4696,
+            "distinct_person_cnt": "1505",
+            "isItem": true,
+            "status": "Def"
+        }
+    ]
+};
+
+// TESTS
+// TODO: What are some good assertions, given various inputs?
+// - makeGraph()
+test('test makeGraph() - diagram case', () => {
+  const uniqueConcepts = [...new Set(diagramCase.flat())];
+  const uniqueConceptObjs = uniqueConcepts.map(d => ({concept_id: d}));
+  const [graph, nodes] = makeGraph(diagramCase, uniqueConceptObjs);
+  
+  // noinspection JSUnresolvedReference idkWhyUnresolved
+    expect(graph.directedSize).toBe(32);
+  expect(graph.nodes()).toStrictEqual(uniqueConcepts);
+});
+
+// TODO:
+test('test makeGraph() - 1 cset "single small"', () => {
+  console.log();
+});
+
+// TODO:
+test('test makeGraph() - 2+ csets', () => {
+  console.log();
 });
 
+// - roots, leaves
+// todo?
+// test('test roots, leaves - diagram case', () => {});
 
-/*
- */
\ No newline at end of file
+test('test roots, leaves - 1 cset "single small"', () => {
```

Comment:
## Test progress: roots, leaves
I don't know if these outputs are correct. That takes more work and possibly your help.

Cases:
- single small
- many small

Assertions
- `expect(gc.roots).toStrictEqual()` - copy/pasted what I got back
- `expect(gc.leaves).toStrictEqual()` - copy/pasted what I got back